nni/algorithms/compression/v2/pytorch/pruning/tools/base.py

```diff
@@ -425,11 +466,22 @@ class TaskGenerator:
         The log directory use to saving the task generator log.
     keep_intermediate_result
         If keeping the intermediate result, including intermediate model and masks during each iteration.
+    best_result_mode
+        The way to decide which one is the best result. Three modes are supported.
+        If the task results don't contain scores (task_result.score is None), it will fall back to ``latest``.
+
+        1. latest: The newest received result is the best result.
+        2. maximize_score: The one with largest task result score is the best result.
```

Comment:
Why `score`? Seems `maximize` and `minimize` are clear enough and aligned with HPO.